department-of-veterans-affairs/va.gov-cms
# VACMS-15785: alt text validation

## Description

Closes #15785.
Closes #15783.

## Testing done
Tested locally. We also developed cypress tests that will verify this won't break in the future. 

## Screenshots
![CleanShot2023-12-01at10 31 06](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/109987005/9dab8572-07f7-43b0-90fc-3dea2ed5ee14)


## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

As a content_editor
1. Navigate to /media/add/image, select an image upload, the alt text field will appear. Enter in alt text using a restricted file extension(.jpg, .gif, .jpeg, .png) and tab or click out of the field
   - [ ] Validate that the error appears
2. Then correct the alt text
   - [ ] Validate that the error disappears
3. Then enter in new alt text with the redundant phrase, i.e. image of, 
   - [ ] Validate the the error appears
4. Then correct the alt text
   - [ ] Validate that the error disappears
5. Then enter in new alt text with the high error count over 150
   - [ ] Validate the the error appears
6. Then correct the alt text
   - [ ] Validate that the error disappears
7. Repeat each step above but click save instead to validate server-side validation, you will need to fill out any other required fields.
   - [ ] Validate error for filename
   - [ ] validate error for redundant phrase
   - [ ] validate error for character count

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
